src/Workspaces/Core/Portable/Shared/Extensions/ITypeSymbolExtensions.ReplaceTypeParameterBasedOnTypeConstraintVisitor.cs

```diff
@@ -98,13 +96,13 @@ public override ITypeSymbol VisitTypeParameter(ITypeParameterSymbol symbol)
                         if (symbol.ConstraintTypes.All(t => t is INamedTypeSymbol))
                         {
                             var immutableProjects = _solution.Projects.ToImmutableHashSet();
-                            var derivedImplementedTypesOfEachConstraintType = symbol.ConstraintTypes.Select(ct =>
+                            var derivedImplementedTypesOfEachConstraintType = (await Task.WhenAll(symbol.ConstraintTypes.Select(async ct =>
                             {
                                 var derivedAndImplementedTypes = new List<INamedTypeSymbol>();
-                                var derivedClasses = SymbolFinder.FindDerivedClassesAsync((INamedTypeSymbol)ct, _solution, immutableProjects, _cancellationToken).WaitAndGetResult(_cancellationToken);
-                                var implementedTypes = DependentTypeFinder.FindTransitivelyImplementingStructuresAndClassesAsync((INamedTypeSymbol)ct, _solution, immutableProjects, _cancellationToken).WaitAndGetResult(_cancellationToken);
+                                var derivedClasses = await SymbolFinder.FindDerivedClassesAsync((INamedTypeSymbol)ct, _solution, immutableProjects, _cancellationToken).ConfigureAwait(false);
+                                var implementedTypes = await DependentTypeFinder.FindTransitivelyImplementingStructuresAndClassesAsync((INamedTypeSymbol)ct, _solution, immutableProjects, _cancellationToken).ConfigureAwait(false);
                                 return derivedClasses.Concat(implementedTypes.Select(s => s.Symbol)).ToList();
-                            }).ToList();
+                            })).ConfigureAwait(false)).ToList();
```

Comment:
would def prefer to introduce locals for `await` calls.  the wrapping with parens and continuing operations is not trivial to read.

Reply:
This should be simpler now (no operations changed on `await` expressions)